radio/util/build-firmware.py

```diff
@@ -219,9 +218,20 @@
         if size > maxsize:
             exit(FIRMWARE_SIZE_TOO_BIG)
 
-
     # Copy binary to the binaries directory
     shutil.move(target, path)
 
+if os.path.isfile(errpath):
+    exit(COMPILATION_ERROR)
+
+lockpath = path + ".lock"
+lock = filelock.FileLock(lockpath)
+try:
+    with lock.acquire(timeout = 60*60):
+        if not os.path.isfile(path):
+            firmware_build(path)
```

Comment:
You defined it L182: build_firmware(path), not firmware_build(path)
I guess it would require filelock to be installed into docker, so a dockerfile modif too